localstack-core/localstack/services/apigateway/next_gen/execute_api/parameters_mapping.py

```diff
@@ -54,39 +60,145 @@ def map_integration_request(
                     method_req_expr, invocation_request
                 )
 
-            elif request_mapping.startswith("context."):
-                context_var_expr = request_mapping.removeprefix("context.")
-                value = self._retrieve_parameter_from_context_variables(
-                    context_var_expr, context_variables
+            else:
+                value = self._retrieve_parameter_from_variables_and_static(
+                    mapping_value=request_mapping,
+                    context_variables=context_variables,
+                    stage_variables=stage_variables,
                 )
 
-            elif request_mapping.startswith("stageVariables."):
-                stage_var_name = request_mapping.removeprefix("stageVariables.")
-                value = self._retrieve_parameter_from_stage_variables(
-                    stage_var_name, stage_variables
+            if value:
+                request_data_mapping[integration_param_location][param_name] = value
+
+        return request_data_mapping
+
+    def map_integration_response(
+        self,
+        response_parameters: dict[str, str],
+        integration_response: Response,
+        context_variables: ContextVariables,
+        stage_variables: dict[str, str],
+    ) -> ResponseDataMapping:
+        response_data_mapping = ResponseDataMapping(header={})
+
+        # storing the case-sensitive headers once, the mapping is strict
+        case_sensitive_headers = dict(integration_response.headers.items())
+
+        for response_mapping, integration_mapping in response_parameters.items():
+            header_name = response_mapping.removeprefix("method.response.header.")
+
+            if integration_mapping.startswith("integration.response."):
+                method_req_expr = integration_mapping.removeprefix("integration.response.")
+                value = self._retrieve_parameter_from_integration_response(
+                    method_req_expr, integration_response, case_sensitive_headers
+                )
+            else:
+                value = self._retrieve_parameter_from_variables_and_static(
+                    mapping_value=integration_mapping,
+                    context_variables=context_variables,
+                    stage_variables=stage_variables,
                 )
 
-            elif request_mapping.startswith("'") and request_mapping.endswith("'"):
-                value = request_mapping.strip("'")
+            if value:
+                response_data_mapping["header"][header_name] = value
+
+        return response_data_mapping
+
+    def _retrieve_parameter_from_variables_and_static(
```

Comment:
Nice! That will be ready for reuse in the gateway responses as well when we get there!